chat-plugins/info.js

```diff
@@ -359,165 +359,200 @@ var commands = exports.commands = {
 		if (!this.canBroadcast()) return;
 
 		if (!target) return this.parse('/help dexsearch');
-		var targets = target.split(',');
-		var searches = {};
+		var searches = [];
 		var allTiers = {'uber':1, 'ou':1, 'bl':1, 'uu':1, 'bl2':1, 'ru':1, 'bl3':1, 'nu':1, 'bl4':1, 'pu':1, 'nfe':1, 'lc uber':1, 'lc':1, 'cap':1};
 		var allColours = {'green':1, 'red':1, 'blue':1, 'white':1, 'brown':1, 'yellow':1, 'purple':1, 'pink':1, 'gray':1, 'black':1};
 		var allStats = {'hp':1, 'atk':1, 'def':1, 'spa':1, 'spd':1, 'spe':1, 'bst':1};
 		var showAll = false;
 		var megaSearch = null;
 		var randomOutput = 0;
-		var categories = ['gen', 'tier', 'color', 'types', 'ability', 'stats', 'compileLearnsets', 'moves', 'recovery', 'priority'];
-
-		for (var i = 0; i < targets.length; i++) {
-			var isNotSearch = false;
-			target = targets[i].trim().toLowerCase();
-			if (target.charAt(0) === '!') {
-				isNotSearch = true;
-				target = target.substr(1);
-			}
 
-			var targetAbility = Tools.getAbility(targets[i]);
-			if (targetAbility.exists) {
-				if (!searches['ability']) searches['ability'] = {};
-				if (Object.count(searches['ability'], true) === 1 && !isNotSearch) return this.sendReplyBox("Specify only one ability.");
-				if ((searches['ability'][targetAbility.name] && isNotSearch) || (searches['ability'][targetAbility.name] === false && !isNotSearch)) return this.sendReplyBox("A search cannot both exclude and include an ability.");
-				searches['ability'][targetAbility.name] = !isNotSearch;
-				continue;
-			}
-
-			if (target in allTiers) {
-				if (!searches['tier']) searches['tier'] = {};
-				if ((searches['tier'][target] && isNotSearch) || (searches['tier'][target] === false && !isNotSearch)) return this.sendReplyBox('A search cannot both exclude and include a tier.');
-				searches['tier'][target] = !isNotSearch;
-				continue;
-			}
+		var self = this;
+		var validParameter = function (cat, param, isNotSearch) {
```

Comment:
As far as I can tell, the only functionality I've removed is more specific error messages because that's a lot more complicated now, but I'm sure someone more clever then myself can figure out how to improve this function to fix that.